t/python/t1000-service-add-remove.py

```diff
@@ -75,6 +79,34 @@ def cb(f, t, msg, arg):
         w2.stop()
         w2.destroy()
 
+    def test_004_service_rpc_user(self):
+        cb_called = [False]  # So that cb_called[0] is mutable in inner function
+        p = {"test": "bar"}
+
+        def cb(f, t, msg, arg):
+            cb_called[0] = True
+            self.assertEqual(msg.payload, p)
+            f.reactor_stop()
+
+        w2 = self.f.msg_watcher_create(cb, FLUX_MSGTYPE_RESPONSE, "foo.echo")
+        w2.start()
+        self.assertIsNotNone(w2, msg="msg_watcher_create response handler")
+
+        m = Message()
+        m.topic = "foo.echo"
+        m.payload = p
+        m.pimpl.set_rolemask(flux.constants.FLUX_ROLE_USER)
+        m.pimpl.set_userid(os.getuid() + 1)
+        self.assertTrue(m is not None)
+        ret = self.f.send(m)
+        self.assertEqual(ret, 0)
+
+        ret = self.f.reactor_run()
+        self.assertTrue(ret >= 0)
+        self.assertTrue(cb_called[0])
+        w2.stop()
+        w2.destroy()
+
```

Comment:
should test opposite case?  handler only allows owner and user is rejected?

Reply:
Great point. I'll add that

Reply:
Ok, that was a bit of work and uncovered some other issues. This PR is now based on top of #5653 since it requires the new `decode` method on the `Message` class.